### PR TITLE
simplify_analysis-III

### PR DESCRIPTION
**Pull request by mschwoer**
simplifies the Analysis and Results page.

Note that I also changed the behaviour: plots are only saved to the "results" page on request.

### CODE REVIEW

File: alphastats/gui/pages/04_Analysis.py [hunk 1]
```diff
@@ -50,90 +33,49 @@ def select_analysis():
...
-        )
+# --- SHOW STATISTICAL ANALYSIS -------------------------------------------------------
+elif show_df:
+    display_df(analysis_result)
 
+    csv = convert_df(analysis_result)
 
-else:
-    st.info("Import Data first")
+    # TODO do we want to save statistical analysis to results page as well?
```
Comment (JuliaS92) [inline alphastats/gui/pages/04_Analysis.py:78]: The answer was yes. Resolve and merge if this was implemented in the meantime.

File: alphastats/gui/utils/analysis_helper.py [hunk 2]
```diff
@@ -75,12 +93,18 @@ def get_unique_values_from_column(column):
 def st_general(method_dict):
     chosen_parameter_dict = {}
 
-    if "settings" in list(method_dict.keys()):
-        settings_dict = method_dict.get("settings")
+    if "settings" not in method_dict:
+        # TODO: is this check really required here? If so, refactor to be part of plotting/statistic_options
```
Comment (JuliaS92) [inline alphastats/gui/utils/analysis_helper.py:97]: I am confused by this, too. Has this resolved in the meantime?

File: alphastats/gui/utils/analysis_helper.py [hunk 3]
```diff
@@ -285,20 +275,19 @@ def helper_plot_dimensionality_reduction(method_dict):
...
     group = st.selectbox(
         "Grouping variable",
-        options=[default_option]
-        + st.session_state[StateKeys.DATASET].metadata.columns.to_list(),
+        options=[default_option] + dataset.metadata.columns.to_list(),
     )
 
     if group != default_option:
-        unique_values = get_unique_values_from_column(column=group)
+        unique_values = dataset.metadata[group].unique().tolist()
```
Comment (JuliaS92) [inline alphastats/gui/utils/analysis_helper.py:290]: Nice!

**Review by JuliaS92 — APPROVED** [review]
Great sorting effort

File: alphastats/gui/utils/analysis_helper.py (hunk 2 above)
Comment (mschwoer) [reply alphastats/gui/utils/analysis_helper.py:97]: yes, this logic is now explicit

**Pull request merged**